### PR TITLE
[backend] Add new needssslcertforbuild regex for Debian builds

### PR DESCRIPTION
**Pull request by bluca**
The only programmatic supported way to add arbitrary strings to a Debian
dsc file is via X[SBC]-Key: value tags in debian/control.
The regex that checks if the SSL certificate needs to be copied into the
build root checks that there is exactly "# needssslcertforbuild", with
the hash sign having to be the first character of the line.
But in Debian at best we can do:
Obs: sign_this # needssslcertforbuild
by adding:
XS-Obs: sign_this #needssslcertforbuild
to debian/control. So relax the regex in bs_worker and remove the need
for the # to be the first character of the line.

### CODE REVIEW

**Comment by bluca** [comment]
I can do that, certainly. Wanted to keep the changes to a minimum - will retest and push in a minute.

**Comment by bluca** [comment]
Done as suggested. I'm working on a debhelper to automate the packaging process, will send a PR to pesign-obs-integration where this new regex will be documented.

File: src/backend/bs_worker [hunk 1]
```diff
@@ -2848,6 +2848,9 @@ sub dobuild {
 	    $kiwiorigins = {};			# but collect origins
 	  }
 	  $needsslcert = 1 if /^#\s*needssslcertforbuild\s*$/s;
+	  # On Debian we cannot add a commented out line in a dsc, use a tag instead
+	  # that can be generated by adding "XS-Obs: needssslcertforbuild" to debian/control
+	  $needsslcert = 1 if /^Obs:\s*needssslcertforbuild\s*$/s;
```
Comment (adrianschroeter) [inline src/backend/bs_worker]: shouldn't it be ^XS-Obs: ... ?

Comment (bluca) [reply src/backend/bs_worker]: @adrianschroeter no, that's how it's "coded", X means user-specified tag and "S" means it's only applied to source artefact (ie: the .dsc file), for example:

```
$ grep Obs debian/control 
XS-Obs: needssslcertforbuild
$ grep Obs ../shim_14-1.dsc 
Obs: needssslcertforbuild
```

The OBS backend parses the .dsc files, not debian/control

**Pull request merged**